### PR TITLE
MicroProfile Metrics 2.3 Implementation

### PR DESCRIPTION
**Pull request by Channyboy**
See #9985 
#build

Implementation includes two new _features_
1. SimpleTimer REST metric
2. Runtime provided REST metrics (from monitoring/tracking REST resources)

### CODE REVIEW

**Comment by Channyboy** [comment]
#build
#run-libby-bot

**Comment by dmuelle** [comment]
Do any of these files require ID review? If not please apply the `ID review not needed` tag. Thanks

**Comment by Channyboy** [comment]
@dmuelle  ID review not needed has been added

**Comment by Channyboy** [comment]
#build
#run-libby-bot

**Comment by Channyboy** [comment]
#build
#run-libby-bot

File: dev/com.ibm.websphere.org.eclipse.microprofile.metrics.2.3/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+Import-Package: \
+  javax.enterprise.util; version="[1.1,3)",\
+  *
+
+Export-Package: \
+  org.eclipse.microprofile.metrics;version=2.3, \
+  org.eclipse.microprofile.metrics.annotation;version=2.3
+
+#Include-Resource: \
+#  @${repo;org.eclipse.microprofile.metrics:microprofile-metrics-api;2.2;EXACT}
```
Comment (fmhwong) [inline dev/com.ibm.websphere.org.eclipse.microprofile.metrics.2.3/bnd.bnd]: This needs to be version 2.3 when it is uncommented.

File: dev/com.ibm.ws.microprofile.metrics.2.3/src/com/ibm/ws/microprofile/metrics23/impl/SimpleTimerImpl.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        }
+    }
+
+    @Override
+    public Context time() {
+        return new Context(this, clock);
+    }
+
+    private void update(long duration) {
+        elapsedTime += duration;
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.2.3/src/com/ibm/ws/microprofile/metrics23/impl/SimpleTimerImpl.java]: Do we need synchronization when using SimpleTimer in multiple threads?

Comment (Channyboy) [reply dev/com.ibm.ws.microprofile.metrics.2.3/src/com/ibm/ws/microprofile/metrics23/impl/SimpleTimerImpl.java]: synchronized added

**Comment by andrea-pichler** [comment]
Requested message review completed by L2

**Comment by Channyboy** [comment]
#build
#run-libby-bot

**Comment by Channyboy** [comment]
Re-based with `master` branch. Will run one more PB.
#build
#run-libby-bot

**Pull request merged**